jps-plugin/src/org/jetbrains/kotlin/compilerRunner/KotlinCompilerRunner.kt

```diff
@@ -0,0 +1,255 @@
+                        compilationCanceledStatus = environment.services.get(CompilationCanceledStatus::class.java))
+
+                val res = KotlinCompilerClient.incrementalCompile(daemon, argsArray, services, compilerOut, daemonOut)
+
+                ProcessCompilerOutput(messageCollector, collector, compilerOut, res.toString())
+                BufferedReader(StringReader(daemonOut.toString())).forEachLine {
+                    messageCollector.report(CompilerMessageSeverity.INFO, it, CompilerMessageLocation.NO_LOCATION)
+                }
+                return true
+            }
+        }
+        return false
+    }
+
+    private fun getReturnCodeFromObject(rc: Any?): String {
+        if (rc == null) {
+            return INTERNAL_ERROR
+        }
+        else if (ExitCode::class.java.name == rc.javaClass.name) {
+            return rc.toString()
+        }
+        else {
+            throw IllegalStateException("Unexpected return: " + rc)
+        }
+    }
+
+    private fun <T : CommonCompilerArguments> mergeBeans(from: CommonCompilerArguments, to: T): T {
+        // TODO: rewrite when updated version of com.intellij.util.xmlb is available on TeamCity
+        try {
+            val copy = XmlSerializerUtil.createCopy(to)
+
+            val fromFields = collectFieldsToCopy(from.javaClass)
+            for (fromField in fromFields) {
+                val toField = copy.javaClass.getField(fromField.name)
+                toField.set(copy, fromField.get(from))
+            }
+
+            return copy
+        }
+        catch (e: NoSuchFieldException) {
+            throw rethrow(e)
+        }
+        catch (e: IllegalAccessException) {
+            throw rethrow(e)
+        }
+
+    }
+
+    private fun collectFieldsToCopy(clazz: Class<*>): List<Field> {
+        val fromFields = ArrayList<Field>()
+
+        var currentClass: Class<*>? = clazz
+        while (currentClass != null) {
+            for (field in currentClass.declaredFields) {
+                val modifiers = field.modifiers
+                if (!Modifier.isStatic(modifiers) && Modifier.isPublic(modifiers)) {
+                    fromFields.add(field)
+                }
+            }
+            currentClass = currentClass.superclass
+        }
+
+        return fromFields
+    }
+
+    private fun setupK2JvmArguments(moduleFile: File, settings: K2JVMCompilerArguments) {
+        settings.module = moduleFile.absolutePath
+        settings.noStdlib = true
+        settings.noJdkAnnotations = true
+        settings.noJdk = true
+    }
+
+    private fun setupK2JsArguments(
+            outputFile: File,
+            sourceFiles: Collection<File>,
+            libraryFiles: List<String>,
+            settings: K2JSCompilerArguments) {
+        settings.noStdlib = true
+        settings.freeArgs = ContainerUtil.map(sourceFiles, object : Function<File, String> {
```

Comment:
Hm, am I supposed to throroughtly rework the code I just converted from Java to Kotlin? 
I understand that it would be nice, but I think if this is the rule - I'd better avoid converting files unless absolutely necessary, because I rewriting them properly takes time.